pyquil/api/_compiler.py

```diff
@@ -403,6 +406,35 @@ def get_quilt_calibrations(self) -> Program:
         calibration_program = parse_program(response.quilt)
         return calibration_program
 
+    @_record_call
+    def refresh_calibrations(self) -> None:
+        """Refresh the calibrations cache."""
+        self._calibrations = self.get_calibrations()
+
+    @property
+    def calibrations(self) -> Program:
+        """Cached calibrations."""
+        if self._calibrations is None:
+            self.refresh_calibrations()
+
+        if self._calibrations is None:
+            raise RuntimeError("Could not refresh calibrations")
+        else:
+            return self._calibrations
+
+    @_record_call
+    def expand_calibrations(self, program: Program, discard_defcals: bool = True) -> Program:
+        # Prepend the system's calibrations to the user's calibrations
+        calibrated_program = self.calibrations + program.copy_everything_except_instructions()
```

Comment:
@kalzoo You'll notice that I switched back to storing the full calibrations program (as provided by `get_quilt_calibrations`) rather than just the `calibrations` property: the program contains other stuff (waveforms, frames) that we don't want to discard. That means we're back to writing stuff like `qc.compiler.calibrations.calibrations` (likewise `qc.compiler.calibrations.waveforms` etc.) but I'm not overly worried about that. The `+` operator for programs handles calibrations, etc., so adding programs works as expected with later calibrations taking precedence.

Alternatively we could have `calibrations`, `waveforms`, and `frames` properties on `QPUCompiler`. That would fix the overloading of the word "calibrations" but otherwise I'm not sure it's an improvement. What do ya think?

Reply:
Ok, sounds good. Let's rename it for clarity then - `calibrations` shouldn't be a program, and `calibrations.calibrations` makes that apparent. `calibrations_program`?